pkg/instance/handler.go

```diff
@@ -32,11 +32,53 @@ type groupServiceHandler interface {
 	Find(ctx context.Context, name string) (*model.Group, error)
 }
 
-type SaveDeploymentRequest struct {
-	Name        string `json:"name" binding:"required,dns_rfc1035_label"`
-	Description string `json:"description"`
-	Group       string `json:"group" binding:"required"`
-	TTL         uint   `json:"ttl"`
+type filestoreBackupRequest struct {
+	// Name of backup
+	Name string `json:"name" binding:"required"`
+}
+
+func (h Handler) FilestoreBackup(c *gin.Context) {
+	// swagger:route POST /instances/{id}/backup filestoreBackup
+	//
+	// Backup filestore
+	//
+	// Backup filestore...
+	//
+	// Security:
+	//	oauth2:
+	//
+	// responses:
+	//	200:
+	//	401: Error
+	//	403: Error
+	//	404: Error
+	//	415: Error
+	id, ok := handler.GetPathParameter(c, "id")
+	if !ok {
+		return
+	}
+
+	var request filestoreBackupRequest
+	if err := handler.DataBinder(c, &request); err != nil {
+		_ = c.Error(err)
+		return
+	}
+
+	ctx := c.Request.Context()
+
+	instance, err := h.instanceService.FindDeploymentInstanceById(ctx, id)
+	if err != nil {
+		_ = c.Error(err)
+		return
+	}
+
+	err = h.instanceService.FilestoreBackup(ctx, instance, request.Name)
+	if err != nil {
+		_ = c.Error(err)
+		return
+	}
+
+	c.Status(http.StatusOK)
```

Comment:
Should this be `201 Created` instead? 🤔

Reply:
It should but I removed the entire handler since we integrating this into our existing save database functionality rather than exposing it as a separate endpoint